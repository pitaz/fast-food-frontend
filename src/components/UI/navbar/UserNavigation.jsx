import React, { Component, Fragment } from "react";
import { Link } from "react-router-dom";
import { bindActionCreators } from 'redux';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';
import logout from '../../../actions/auth/logout';

class UserNavigation extends Component {
  logout = () => {
    const { logout } = this.props;
    logout();
  };

  render() {
    const { role } = this.props;
    return (
     <Fragment>
       {role === 'user'}
       <Link 
       to='/'
       id='logout'
       onClick={this.logout}>Logout</Link>
     </Fragment>
    );
  }
}
UserNavigation.propTypes = {
 role: PropTypes.string,
 logout: PropTypes.func
};

const mapStateToProps = state => ({
role: state.auth.user.role
});

const mapDispatchToProps = dispatch => bindActionCreators(
  {
    logout: logout
  },
  dispatch
);
export default connect(mapStateToProps, mapDispatchToProps)(UserNavigation);
