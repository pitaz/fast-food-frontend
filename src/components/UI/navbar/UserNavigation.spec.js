/* eslint-disable no-undef */
import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import UserNavigation from './UserNavigation.jsx';


describe('UserNavigation Component', () => {
  const wrapper = shallow(<UserNavigation />);
  it('should render the UserNavigation component', () => {

    expect(wrapper.exists()).toBe(true);
  });

  it('should render the UserNavigation component for users', () => {
    const props = {
      logoutUser: jest.fn(),
      userRole: 2
    };

    // const wrapper = shallow(<UserNavigation {...props} />);

    expect(wrapper.exists()).toBe(true);
  });

  it('should render the UserNavigation component for admin', () => {
    const props = {
      logoutUser: jest.fn(),
      userRole: 1
    };

    // const wrapper = shallow(<UserNavigation {...props} />);

    expect(wrapper.exists()).toBe(true);
  });

  it('should match the snapshot', () => {
    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
