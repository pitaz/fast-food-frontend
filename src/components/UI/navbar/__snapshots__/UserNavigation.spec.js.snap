// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UserNavigation Component should match the snapshot 1`] = `
<ContextConsumer>
  <Component />
</ContextConsumer>
`;
